sling/nlp/wikicat/app/wikicat.js

```diff
@@ -29,10 +30,15 @@ app.controller('SearchCtrl', function($scope, $http,  $location, $mdDialog) {
       $scope.num_categories = response.data["num_categories"];
   });
 
-  $scope.query = "";                // main text box
-  $scope.query_results = null;      // results of the query
-  $scope.recordio_results = null;   // result of recordio generation
-  $scope.parse_selector = "";       // user-specified parse-selection formula
+  $scope.query = "";                   // main text box
+  $scope.query_results = null;         // results of the query
+  $scope.recordio_results = null;      // result of recordio generation
+  $scope.parse_selector = "";          // user-specified parse-selection formula
+  $scope.selected_parses = new Set();  // user-selected parses
+
+  // Current page and page size for displaying results.
+  $scope.current_page = 0;
+  $scope.page_size = 50;
```

Comment:
Just wondering if it would make sense to use 200 here as well.

Reply:
I have increased it to 200, and the response size from the server to 500. I think we might hit the limits of AngularJS beyond that :-) For one, the recomputation cost after every UI change might be much bigger. Also, the size of the requests to the server might exceed the GET limits (~2K chars).
Please feel free to reduce this (and the server response size) if you face any problems.